File: eventbus/eclipse_cpp/eventbus/src/unittest/LoggerExample.cpp

#include <LoggerExample.hpp>

   LoggerExample::LoggerExample() {

   }

   void LoggerExample::sinkEvent(EBEvent event) {

      rcvQueue.push(event);
   }

   bool LoggerExample::hasEvents() {

      return rcvQueue.size() > 0;
   }

   void LoggerExample::processEvents() {

      int size = rcvQueue.size();

      for (int i = 0; i < size; i++) {

         EBEvent event;

         if (rcvQueue.front(event)) {

            std::cout << "\nEvent info:" << std::endl;
            std::cout << "publisherID = " << event.publisherID << std::endl;
            std::cout << "eventID = " << event.eventID << std::endl;
            std::cout << "eventCatType = "<< event.eventCatType << std::endl;
            std::cout << "eventDefType = " << event.eventDefType << std::endl;
            std::cout << "eventData = " << event.eventData << std::endl;
         }
      }
   }

File: eventbus/eclipse_java/eventbus/src/orbisoftware/msgblit/eventbus/unittest/LoggerExample.java
package orbisoftware.msgblit.eventbus.unittest;

import java.util.concurrent.ConcurrentLinkedQueue;
import java.util.Queue;

import orbisoftware.msgblit.eventbus.EBEvent;
import orbisoftware.msgblit.eventbus.EBSubscriber;

public class LoggerExample implements EBSubscriber {

   LoggerExample() {

      rcvQueue = new ConcurrentLinkedQueue<EBEvent>();
   }

   @Override
   public void sinkEvent(EBEvent event) {

      if (rcvQueue.size() < maxQueueSize)
         rcvQueue.add(event);
   }

   public boolean hasEvents() {

      return rcvQueue.size() > 0;
   }

   public void processEvents() {

      int size = rcvQueue.size();

      for (int i = 0; i < size; i++) {

         EBEvent event = rcvQueue.poll();

         if (event != null) {

            System.out.println("\nEvent info:");
            System.out.println("publisherID = " + event.publisherID);
            System.out.println("eventID = " + event.eventID);
            System.out.println("eventCatType = " + event.eventCatType);
            System.out.println("eventDefType = " + event.eventDefType);
            System.out.println("eventData = " + event.eventData);
         }
      }
   }

   private Queue<EBEvent> rcvQueue;
   private int maxQueueSize = 100;
}
